### PR TITLE
minor refactor in torrent_status and minor header cleanup in torrent_handle

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: include/libtorrent/torrent_status.hpp [hunk 1]
```diff
@@ -545,7 +546,7 @@ namespace libtorrent
 		bool stop_when_ready = false;
 
 		// the info-hash for this torrent
-		sha1_hash info_hash{nullptr};
```
Comment (arvidn) [inline include/libtorrent/torrent_status.hpp]: ? I seem to recall that the default constructor leaves the ``sha1_hash`` uninitialized.

Comment (aldenml) [inline include/libtorrent/torrent_status.hpp]: no :) (https://github.com/arvidn/libtorrent/blob/master/include/libtorrent/sha1_hash.hpp#L76)

Comment (arvidn) [inline include/libtorrent/torrent_status.hpp]: I see. passing in ``nullptr`` should probably be prohibited then...

**Comment by arvidn** [comment]
lgtm

**Pull request merged**